### PR TITLE
feat: add `ce:review-beta` with structured persona pipeline

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

- Adds `ce:review-beta`, a new beta review skill that replaces ce:review's free-form agent dispatch with a structured 6-stage pipeline adapted from the [iterative-engineering plugin's code-review skill](https://github.com/tmchow/tmc-marketplace)
- Adds 8 new persona agents with confidence calibration, false-positive suppression, and structured JSON output
- Integrates CE-specific agents (agent-native-reviewer, learnings-researcher, schema-drift-detector, deployment-verification-agent) into the pipeline

## Why this is different from ce:review

The existing `ce:review` dispatches autonomous agents (security-sentinel, performance-oracle, etc.) that return **unstructured prose reports**. The orchestrator collects these reports and synthesizes them for the user.

In practice, this produces reviews where real bugs get buried in noise. An agent told to "be paranoid and leave no stone unturned" will generate 15 findings, but only 3 are real problems -- the rest are vague suggestions like "consider adding rate limiting" or defensive coding advice for values that can't actually be null. With 5+ agents running, the same unbounded query gets flagged by both the security and performance agents with different framing, so the report looks twice as long as it should. And every finding looks equally important because there's no way to distinguish "this will crash in production" from "this might theoretically be an issue at extreme scale." The user ends up manually triaging the review output itself, which defeats the purpose of automated review.

`ce:review-beta` fixes this with pipeline-level quality control:

| | ce:review | ce:review-beta |
|-|-----------|----------------|
| **Agent output** | Free-form prose (Executive Summary, Risk Matrix, etc.) | Structured JSON matching a schema (severity, confidence, evidence, file:line) |
| **False positive control** | None -- agents told to "be paranoid, leave no stone unturned" | Each persona has explicit "what you don't flag" anti-patterns and per-domain confidence thresholds |
| **Confidence scoring** | None -- every finding looks equally important | 0.0-1.0 per finding, calibrated per domain (security at 0.60 = actionable; performance at 0.60 = noise). Findings below threshold are suppressed before the user sees them |
| **Multi-agent dedup** | None -- overlapping findings from different agents appear as duplicates | Fingerprint-based merge: same file + line range + issue = single finding with highest severity, unioned evidence |
| **Agent selection** | User-configured roster via settings file | Tiered: 3 always-on + 5 conditional selected by diff content |
| **Intent awareness** | None -- agents review code in a vacuum | Intent discovery step before spawning; shapes how hard each reviewer looks |
| **Quality gates** | None | Verify actionability, ban vague language ("consider improving..."), check line accuracy, calibrate severity |

## Review modes

The skill supports three execution modes, chosen by passing a mode token in the arguments:

| Mode | Trigger | Behavior |
|------|---------|----------|
| **Interactive** (default) | No mode token | Asks the user one question to establish intent before spawning reviewers. Presents findings and offers to fix. |
| **Autonomous** | `mode:autonomous` | No user interaction. Runs the full pipeline, applies only policy-allowed `safe_auto` fixes, re-reviews in bounded rounds, writes a run artifact, and emits residual work as `file-todos` for downstream resolution. |
| **Report-only** | `mode:report-only` | Strictly read-only. Reviews and reports findings, then stops. No edits, todos, commits, pushes, or PR actions. Safe for parallel execution on the same checkout (e.g., alongside browser testing). Will refuse to switch the shared checkout if given a PR or branch target -- must run in an isolated worktree or on the already checked-out branch. |

The mode design supports pipeline composition: an orchestrator can run `mode:report-only` in parallel with other tools on the same checkout, then follow up with `mode:autonomous` in a dedicated worktree for fixes.

## The persona agents

8 new agents in `agents/review/`, each with:
- **Confidence calibration** with per-domain thresholds (security has a lower bar because the cost of missing a vuln is high; performance has a higher bar because perf issues are easy to measure later)
- **"What you don't flag"** sections that prevent common LLM false positives (e.g., security reviewer won't suggest "consider adding rate limiting" without a specific exploitable finding)
- **`tools: Read, Grep, Glob, Bash`** frontmatter for read-only enforcement
- **Structured JSON output** matching `findings-schema.json`

| Persona | Tier | Focus |
|---------|------|-------|
| correctness-reviewer | Always-on | Logic errors, edge cases, state bugs, error propagation |
| testing-reviewer | Always-on | Coverage gaps, weak assertions, brittle tests |
| maintainability-reviewer | Always-on | Coupling, complexity, naming, dead code |
| security-reviewer | Conditional | Exploitable vulnerabilities (attacker mindset, not compliance checklist) |
| performance-reviewer | Conditional | Runtime perf and scalability (N+1, unbounded memory, blocking I/O) |
| api-contract-reviewer | Conditional | Breaking changes to public interfaces |
| data-migrations-reviewer | Conditional | Migration safety, swapped IDs, dual-write, orphaned references |
| reliability-reviewer | Conditional | Failure modes, missing timeouts, error swallowing, retry storms |

The `data-migrations-reviewer` was enhanced beyond the iterative-engineering source with items from CE's existing `data-migration-expert` (swapped ID/enum mapping detection, dual-write validation, orphaned reference search).

## CE-specific adaptations

On top of the iterative-engineering base:
- `agent-native-reviewer` and `learnings-researcher` as always-on CE agents
- `schema-drift-detector` and `deployment-verification-agent` as conditional (migration PRs)
- Protected artifacts: findings against `docs/brainstorms/`, `docs/plans/`, `docs/solutions/` are discarded
- Quality gates from structured review best practices

## Test plan

- [ ] Run `/ce:review-beta` on a PR with code changes -- verify structured output with confidence scores
- [ ] Run `/ce:review-beta` on a PR with migration files -- verify conditional agents (data-migrations, schema-drift-detector) are selected
- [ ] Run `/ce:review-beta` with no argument on a feature branch -- verify standalone mode works
- [ ] Run `/ce:review-beta <branch-name>` for a branch without a PR -- verify branch mode doesn't fail
- [ ] Run `/ce:review-beta mode:autonomous` -- verify no user interaction, safe_auto fixes applied, residual todos created
- [ ] Run `/ce:review-beta mode:report-only` -- verify strictly read-only, no edits or mutations
- [ ] Run `/ce:review-beta mode:report-only <PR-number>` -- verify it refuses to switch the shared checkout
- [ ] Verify `bun run release:validate` passes

### CODE REVIEW

File: plugins/compound-engineering/skills/ce-review-beta/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+**If a branch name is provided as an argument:**
+
+Check out the named branch, then diff it against the base branch. Substitute the provided branch name (shown here as `<branch>`):
+
+```
+git checkout <branch>
+```
+
+```
+BASE=$(git merge-base HEAD $(git rev-parse --verify origin/main 2>/dev/null || git rev-parse --verify main 2>/dev/null || echo master)) && echo "BASE:$BASE" && echo "FILES:" && git diff --name-only ${BASE} && echo "DIFF:" && git diff -U10 ${BASE}
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-review-beta/SKILL.md]: Valid. Fixed in 580e960 -- branch mode now checks `git status --porcelain` before `git checkout`. If the worktree is dirty, the agent informs the user to stash/commit first or use PR mode instead. No checkout proceeds until the worktree is clean.

**Comment by tmchow** [comment]
@codex review again.

File: plugins/compound-engineering/skills/ce-review-beta/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+```
+gh pr view <number-or-url> --json title,body,baseRefName,headRefName
+```
+
+Then compute a local diff against the PR's base branch so re-reviews also include local fix commits and uncommitted edits. Substitute the PR base branch from metadata (shown here as `<base>`). If the base ref is not present locally after `gh pr checkout`, fetch it before computing the merge-base:
+
+```
+PR_BASE_REF=$(git rev-parse --verify origin/<base> 2>/dev/null || git rev-parse --verify <base> 2>/dev/null || true)
+if [ -z "$PR_BASE_REF" ]; then
+  git fetch --no-tags origin <base>:refs/remotes/origin/<base> 2>/dev/null || git fetch --no-tags origin <base> 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-review-beta/SKILL.md]: > Resolve PR bases from the base repo instead of always `origin`

Addressed in `5c74d8b`. PR mode now derives the base repository from the PR URL returned by `gh pr view`, resolves an existing local remote that points at that repo when present, and otherwise fetches the base branch directly from `https://github.com/<base-repo>.git` instead of assuming `origin`. The local diff is then computed against that resolved base ref.

File: plugins/compound-engineering/skills/ce-review-beta/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+    if git rev-parse --verify "origin/$candidate" >/dev/null 2>&1 || git rev-parse --verify "$candidate" >/dev/null 2>&1; then
+      REVIEW_BASE_BRANCH="$candidate"
+      break
+    fi
+  done
+fi
+if [ -n "$REVIEW_BASE_BRANCH" ] && ! git rev-parse --verify "origin/$REVIEW_BASE_BRANCH" >/dev/null 2>&1 && ! git rev-parse --verify "$REVIEW_BASE_BRANCH" >/dev/null 2>&1; then
+  git fetch --no-tags origin "$REVIEW_BASE_BRANCH:refs/remotes/origin/$REVIEW_BASE_BRANCH" 2>/dev/null || git fetch --no-tags origin "$REVIEW_BASE_BRANCH" 2>/dev/null || true
+fi
+if [ -n "$REVIEW_BASE_BRANCH" ]; then BASE=$(git merge-base HEAD $(git rev-parse --verify origin/$REVIEW_BASE_BRANCH 2>/dev/null || git rev-parse --verify $REVIEW_BASE_BRANCH 2>/dev/null) 2>/dev/null) || BASE=""; else BASE=""; fi
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-review-beta/SKILL.md]: > Resolve branch-mode diff base from the PR base repository

Addressed: Branch-mode and standalone-mode base resolution now mirrors PR-mode logic -- when `gh pr view` returns an open PR, the base ref is resolved from the PR's actual base repository (extracted from the PR URL), not hard-coded to `origin`. Falls back to `origin` only when no PR exists or the PR base repo has no matching remote.
